src/Couchbase.Lite.Shared/Query/XQuery.cs

```diff
@@ -70,18 +70,27 @@ internal class XQuery : QueryBase
 
         #region Protected Methods
 
-        protected void Copy(XQuery source)
+        protected void Move(XQuery source)
```

Comment:
I don't think I have the full picture to understand what will happen. If I have something like this (ignore the exact type name and syntax) :

From from = select(...).from(DataSource.Collection(collection))

Where where 1 = from.where(....)
Where where 2 = from.where(....)

Will there where2 has a valid data source?

Reply:
Hmmm no, it won't, but this situation will never work here:

If I don't "move" then disposing where1 will invalidate where2.  If I do, then where2 cannot be created....